### ISSUE TITLE
<Esc> send a temporary prefix for root table keys

### ISSUE THREAD

**Issue by alone8172** [tmux/tmux#1231]
## Environment
- tmux version: **tmux 2.1**
- Platform: **Linux Mint 18.2 Sonya x86_64**
- Terminal emulator: **konsole**
- Terminal: **bash**
- $TERM outside tmux: **xterm**
- $TERM inside tmux: **screen**

### Briefly
If bind some keys to root table(`-n` or `-T root`) *Esc* or *ctrl-[* will not directly pass to terminal(**bash**) but hold by **tmux** for a short time. (Using **Vim** with showcmd can observe *Esc* is hold by tmux.  press *Esc*, showing `^[` is postponed for a while. `^[` is shown immediately when **Vim** run directly in **bash**)

### Step by Step to Reproduce this Issue
0. this was tested without a *tmux.conf* file
1. run **bash**
2. run **tmux**
3. *prefix* then `:bind -n M-j display "keystroke detect"`
4. press *Esc* then press *j*
5. one will see a prompt("keystroke detect") on bottom of tmux

**Comment by nicm** [comment]
What are you asking?

**Comment by nicm** [comment]
tmux needs to wait for a while after it sees Escape to see if you are really pressing M-j. You can reduce the time it waits by changing escape-time.

**Comment by alone8172** [comment]
@nicm 
"tmux needs to wait for a while after it sees Escape to see if you are really pressing M-j."
<Esc> was not *prefix*, right? Why tmux need to see if I'm gonna press *M-j*?
More that that, I just press *j* instead of *M-j*. The "keystroke detect" message is shown.
Or do I miss understand you or functionality of tmux?

**Comment by alone8172** [comment]
Sorry to you all. This was even not a issue. I feel really sorry.
I've find the instruction of *Esc* by following keyword **escape-time** given by @nicm .
> escape-time time
   Set the time in milliseconds for which tmux waits after an escape is input to determine if it is  part of a function or meta key sequences.  The default is 500 milliseconds.
